#**Crawling Komentar Opini Teknologi**
Judul Vidio: **Hati - hati Pengangguran Karena AI** <br>
Channel Youtube: **Raymond Chin**

**Import Library**

In [1]:
import pandas as pd
from googleapiclient.discovery import build

**Fungsi Crawling Komentar Youtube**

In [10]:
def get_youtube_comments(video_id, api_key):
    # Membangun objek layanan YouTube API v3 dengan kunci API yang diberikan
    youtube = build('youtube', 'v3', developerKey=api_key)
    # List kosong untuk menyimpan semua data komentar (top-level dan balasan)
    comments_data = []

    # Membuat permintaan awal ke YouTube API untuk mendapatkan thread komentar
    request = youtube.commentThreads().list(
        part='snippet,replies', # Meminta 'snippet' untuk detail komentar dan 'replies' untuk balasan
        videoId=video_id, # ID video YouTube yang akan dicrawl
        textFormat='plainText', # Meminta teks komentar dalam format teks biasa
        maxResults=100 # Jumlah maksimal hasil per permintaan (API limit)
    )

    # Loop untuk mengambil semua halaman komentar (pagination)
    while request:
        # Mengeksekusi permintaan API dan mendapatkan respons
        response = request.execute()
        # Iterasi melalui setiap item (thread komentar) dalam respons
        for item in response['items']:
            # Mengakses snippet komentar level teratas (komentar utama)
            top_comment = item['snippet']['topLevelComment']['snippet']
            # Menambahkan data komentar utama ke list
            comments_data.append({
                'date': top_comment['publishedAt'], # Tanggal publikasi komentar
                'username': top_comment['authorDisplayName'], # Nama pengguna yang berkomentar
                'comment_text': top_comment['textDisplay'], # Isi teks komentar
                'is_reply': False, # Menandai bahwa ini bukan balasan
                'reply_to': None # Tidak ada balasan karena ini komentar utama
            })

            # Cek apakah ada balasan untuk komentar utama ini
            if 'replies' in item:
                # Iterasi melalui setiap balasan dalam thread komentar
                for reply in item['replies']['comments']:
                    # Mengakses snippet balasan komentar
                    reply_snippet = reply['snippet']
                    # Menambahkan data balasan komentar ke list
                    comments_data.append({
                        'date': reply_snippet['publishedAt'], # Tanggal publikasi balasan
                        'username': reply_snippet['authorDisplayName'], # Nama pengguna yang membalas
                        'comment_text': reply_snippet['textDisplay'], # Isi teks balasan
                        'is_reply': True, # Menandai bahwa ini adalah balasan
                        'reply_to': top_comment['authorDisplayName'] # Nama pengguna yang dibalas
                    })

        # Mengambil permintaan berikutnya jika ada (untuk pagination)
        request = youtube.commentThreads().list_next(request, response)

    # Mengembalikan data komentar (utama dan balasan) dalam bentuk DataFrame pandas
    return pd.DataFrame(comments_data)

**Jalankan Crawling**

In [11]:
# Sesuaikan dengan id Youtube yang ini di crawl
video_id = "aey4xkrz7zE"

# Kunci API Youtube v3
api_key = "AIzaSyDyNgOhCYlhjuUa9TPiglDVSThXz1MxHG4"

# Memanggil fungsi untuk mendapatkan komentar
comments_df = get_youtube_comments(video_id, api_key)

# Menampilkan 5 baris pertama dari DataFrame komentar
print("Jumlah total komentar:", len(comments_df))
comments_df.head()

Jumlah total komentar: 1055


date               username  \
0  2025-11-13T01:33:25Z         @MsIrawati-v9h   
1  2025-11-13T01:32:26Z         @MsIrawati-v9h   
2  2025-11-11T03:05:54Z  @edysrihermayudha6697   
3  2025-10-31T04:09:21Z    @darmawandesign7820   
4  2025-10-28T06:56:32Z          @DJOE_TOKDJOE   

                                        comment_text  is_reply reply_to  
0  Lama lama manusia di ganti robot...tidak berma...     False     None  
1  AI kalau boleh tidak usah di kembangkan di Ind...     False     None  
2  AI klo kita sebagai manusia tidak memanfaatkan...     False     None  
3  ⛔AI sekarang lebih canggih... bisa menyerap iD...     False     None  
4  Bisa gak Hacker2 jago.. menghancurkan A.I itu ...     False     None

**Simpan Dalam Bentuk Csv**

In [12]:
# Menyimpan DataFrame 'comments_df' ke file CSV
comments_df.to_csv('youtube_comments.csv', index=False) # 'index=False' mencegah penyimpanan indeks DataFrame sebagai kolom
# Mencetak pesan konfirmasi setelah penyimpanan berhasil
print("Data komentar telah berhasil disimpan ke youtube_comments.csv")

Data komentar telah berhasil disimpan ke youtube_comments.csv


Tidak semuanya tercrawling mungkin terdapat batasan dari API per kuota pengambilan. Dari 1.289 komentar yang dapat diambil datanya hanya 1055